File: golang/day1/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func parseInput() []int {
	var numbers []int
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		value, _ := strconv.Atoi(scanner.Text())
		numbers = append(numbers, value)
	}
	return numbers
}

func main() {
	var netValue int
	var repeatedValue int
	var freq = make(map[int]uint)

	var numbers = parseInput()

outer:
	for iter := 0; ; iter++ {

		for _, value := range numbers {
			netValue += value

			freq[netValue]++
			if freq[netValue] > 1 {
				repeatedValue = netValue
				break outer
			}
		}

		if iter == 0 {
			fmt.Println("Result value:", netValue)
		}

	}

	fmt.Println("Repeated value:", repeatedValue)

}

File: rust/day1/src/main.rs
use std::io::{self, BufRead};
use std::collections::HashMap;


fn parse_input() -> Vec<isize> {
    let mut numbers = Vec::new();
    let stdin = io::stdin();

    for line in stdin.lock().lines() {
        let val = line.unwrap();
        numbers.push(val.parse::<isize>().unwrap());
    }

    numbers
}

fn main() {

    let mut iter = 0;
    let mut net_val = 0;
    let mut repeated_value: Option<isize> = None;
    let mut freq: HashMap<isize, usize> = HashMap::new();

    let numbers = parse_input();

    while repeated_value == None {        
        for value in &numbers {
            
            net_val += value;
            
            *freq
            .entry(net_val)
            .or_insert(0) += 1;

            if freq[&net_val] > 1 {                
                repeated_value = Some(net_val);
                break;
            }
        }

        if iter == 0 {
            println!("Result value: {}", net_val);
        }

        iter += 1;
    }

    println!("Repeated value: {}", repeated_value.unwrap());

}
